relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -86,6 +128,24 @@ impl GlobCondition {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CustomCondition {
+    pub name: String,
+    #[serde(default)]
+    pub value: Value,
```

Comment:
If we have generic options, why do we need `value` here?

For a follow-up: It would be interesting to type out the custom conditions, since they are statically known in the code.

Reply:
Understood. Once these are typed out it certainly looks better again. Let's keep like this, then.

Reply:
On another thought: It seems odd that we have `options` but `name` and `value` special cased on the top, once everything is typed out. We can address this when typing out, though.